### PR TITLE
Fix xxHash64 handling of large (> 4GB) inputs

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
Contributes to https://github.com/dotnet/runtime/issues/69184.

Our implementation of xxHash64 stored the length as a 32-bit integer internally. This meant that we're producing incorrect hashes when handling inputs greater than 4GB, as we're losing some of the length bits that should be folded into the final digest.

I couldn't find any real test cases for this scenario, so I ran 10GB worth of "EEEEEE..." through the reference implementation and observed its outputs given the various seeds we use in our test infrastructure. After changing our length handling to be 64-bit, our outputs match the reference implementation.

### CODE REVIEW

File: src/libraries/System.IO.Hashing/tests/XxHash64Tests.007.cs [hunk 1]
```diff
@@ -114,6 +114,31 @@ public static IEnumerable<object[]> TestCases
...
+                    arr[0] = testCase;
+                    yield return arr;
+                }
+            }
+        }
+
+        protected static IEnumerable<LargeTestCase> LargeTestCaseDefinitions { get; } =
+            new[]
+            {
+                //https://asecuritysite.com/encryption/xxHash, Example 1
```
Comment (bartonjs) [inline src/libraries/System.IO.Hashing/tests/XxHash64Tests.007.cs:134]: I see example 1 as

`Try "Nobody inspects the spammish repetition". Try!, which should give a hash of: 0xe2293b2f`

File: src/libraries/System.IO.Hashing/tests/XxHash32Tests.007.cs [hunk 2]
```diff
@@ -92,6 +92,31 @@ public static IEnumerable<object[]> TestCases
...
+
+        protected static IEnumerable<LargeTestCase> LargeTestCaseDefinitions { get; } =
+            new[]
+            {
+                //https://asecuritysite.com/encryption/xxHash, Example 1
+                new LargeTestCase(
+                    "EEEEE... (10GB)",
+                    (byte)'E',
+                    10L * 1024 * 1024 * 1024, // 10 GB
+                    "1C44F650"),
```
Comment (bartonjs) [inline src/libraries/System.IO.Hashing/tests/XxHash32Tests.007.cs:117]: Just pedantically double-checking:  We match the reference implementation here?

Just making sure they didn't shut down the accumulator when the length rolled past either 31 or 32 bits.

**Review by bartonjs — APPROVED** [review]
LGTM, other than some comments describing the test cases that don't seem to make sense

File: src/libraries/System.IO.Hashing/tests/XxHash32Tests.007.cs (hunk 2 above)
Comment (GrabYourPitchforks) [reply src/libraries/System.IO.Hashing/tests/XxHash32Tests.007.cs:117]: Correct, I ran our inputs against the reference implementation for both 32 and 64 bit variants. The reference implementation appears to simply integer overflow, same as we do.

**Comment by GrabYourPitchforks** [comment]
Thanks for the review! :)

**Pull request merged**